### PR TITLE
Run scheduled events at the same time for interval types greater than hours

### PR DESCRIPTION
**Pull request by davidebukali**
#421

### CODE REVIEW

**Review by edwinmp — CHANGES_REQUESTED** [review]
I don't think u need to complicate it to handle illogical configs. It's as simple as each SE should have its start time calculated from the `start time` of the previous instance & the `configured interval`.
If user decides to be unrealistic & configures a SE to run every 30mins, yet each run lasts 2hrs, that's on them. We've already handled that by preventing concurrent runs of the same event.
What I expect here is to move the `create_next_run_instance` to when the current one starts.
Perhaps the only other validation you can do is to make sure that if the `Expected runtime` is configured, then be sure to set the new instance to start after the current one is expected to end.

File: data/management/commands/run_schedules.py [hunk 1]
```diff
@@ -114,28 +114,42 @@ def run_and_update_schedule(self, schedule, runInstance):
...
-                minute=instance_start_date.minute,
-                second =instance_start_date.second
+        updated_runtime = next_runtime.replace(
+            hour=instance_start_date.hour,
+            minute=instance_start_date.minute,
+            second =instance_start_date.second
+        )
+        return updated_runtime
+
+    def is_updated_runtime_less_than_expected_runtime(self, updated_runtime, current_runInstance, schedule):
```
Comment (edwinmp) [inline data/management/commands/run_schedules.py]: This naming implies that this returns a boolean...

**Review by edwinmp — CHANGES_REQUESTED** [review]

File: data/management/commands/run_schedules.py (hunk 1 above)
Comment (davidebukali) [reply data/management/commands/run_schedules.py]: Updated function naming

File: data/management/commands/run_schedules.py [hunk 2]
```diff
@@ -113,11 +113,42 @@ def run_and_update_schedule(self, schedule, runInstance):
...
+            updated_runtime = self.update_next_runtime_to_previous_start_time(schedule, runInstance.start_at, make_aware(datetime.now()))
+            actual_runtime = self.get_most_recent_of_updated_runtime_or_expected_runtime(updated_runtime, runInstance, schedule)
+            self.create_next_run_instance(schedule, actual_runtime)
+
+    def update_next_runtime_to_previous_start_time(self, schedule, instance_start_date, current_date_time):
+        next_runtime = self.calculate_next_runtime(
+            current_date_time,
+            schedule.interval,
+            schedule.interval_type
+        )
```
Comment (edwinmp) [inline data/management/commands/run_schedules.py]: Correct me if I'm wrong, but I think we only need this for this function... just needs to replace `current_date_time` on line 127 with `instance_start_date`

**Review by edwinmp — COMMENTED** [review]
Sorry to be giving you fragmented reviews ... will try to be thorough from the start next time.

**Review by edwinmp — DISMISSED** [review]
If you're sure...

File: data/management/commands/run_schedules.py (hunk 2 above)
Comment (davidebukali) [reply data/management/commands/run_schedules.py]: You are right, using `instance_start_date` ensures we have the same start time for full cycle intervals (greater than 24hrs).

**Review by edwinmp — APPROVED** [review]

**Pull request merged**